### PR TITLE
Add the ability to disable remote-sim tests, since not all build/host machines

### PR DESCRIPTION
**Pull request by schweitzpgi**
have the resources to perform such tests.

### CODE REVIEW

**Comment by khalatepradnya** [comment]
Q: What about Python test(s) in `python/tests/remote/test_remote_platform.py` ?

**Comment by schweitzpgi** [comment]
> Q: What about Python test(s) in `python/tests/remote/test_remote_platform.py` ?

I already had them disabled. 🐹 

Let me take a look...

**Comment by khalatepradnya** [comment]
> > Q: What about Python test(s) in `python/tests/remote/test_remote_platform.py` ?
> 
> I already had them disabled. 🐹
> 
> Let me take a look...

Okay... I see that's outside of this PR.

File: CMakeLists.txt [hunk 1]
```diff
@@ -81,7 +81,16 @@ endif()
 
 # Enable the REST code by default.
 if (NOT DEFINED CUDAQ_ENABLE_REST)
-  set(CUDAQ_ENABLE_REST ON)
+  set(CUDAQ_ENABLE_REST ON CACHE BOOL "Enable building REST client & server.")
+endif()
+
+# Enable the remote simulator by default.
+if (CUDAQ_ENABLE_REST AND NOT DEFINED CUDAQ_ENABLE_REMOTE_SIM)
+  set(CUDAQ_ENABLE_REMOTE_SIM ON CACHE BOOL "Enable building quake-qpud.")
```
Comment (1tnguyen) [inline CMakeLists.txt:89]:
```suggestion
  set(CUDAQ_ENABLE_REMOTE_SIM ON CACHE BOOL "Enable building cudaq-qpud.")
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -81,7 +81,16 @@ endif()
...
+  # Optionally enable the tests that use quake-qpud.
```
Comment (1tnguyen) [inline CMakeLists.txt:90]:
```suggestion
  # Optionally enable the tests that use cudaq-qpud.
```

**Review by 1tnguyen — APPROVED** [review]
👍

**Pull request merged**